### PR TITLE
Add plain text filter operator to escape special chars

### PR DESCRIPTION
**Pull request by Christian862**
## Description
Search input would generally use the `FilterOperator.Like` , which is the regex filter operator. Because of this, you were unable to search with special characters, because they were not escaped. This PR introduces the `FilterOperator.Contains` in order to escape the special characters and treat the string as plain text.

In this PR, we look for the `FilterOperator.Contains` filters and do two things:
1. Transform the value of the filter to escape the special character's, allowing the GQL to treat it as plain text
2. Change the operator type from: `FilterOperator.Contains` to: `FilterOperator.Like`
  - This is because the backend doesn't support a plain text field yet, so we transform the regex string, to act like plain text, then pass it to the query as a `FilterOperator.Like`.  This way In the future when backend support is in for this, we just correct the mapping and remove the string transformation and its working.

### Testing
Please describe the tests that you ran to verify your changes. Please summarize what did you test and what needs to be tested e.g. deployed and tested helm chart locally. 

### Checklist:
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

### Documentation
Make sure that you have documented corresponding changes in this repository or [hypertrace docs repo](https://github.com/hypertrace/hypertrace-docs-website) if required.

### CODE REVIEW

File: projects/observability/src/shared/services/filter-builder/graphql-filter-builder.service.ts [hunk 1]
```diff
@@ -37,10 +38,16 @@ export class GraphQlFilterBuilderService {
         new GraphQlFieldFilter(
           { key: filter.field, subpath: filter.subpath },
           toGraphQlOperator(filter.operator),
-          filter.value as GraphQlArgumentValue
+          this.transformValueIfOperatorIsContains(filter)
         )
     );
   }
+
+  private transformValueIfOperatorIsContains(filter: Filter | FieldFilter | TableFilter): GraphQlArgumentValue {
```
Comment (aaron-steinfeld) [inline projects/observability/src/shared/services/filter-builder/graphql-filter-builder.service.ts:46]: nit: would rename to just something like "extractGraphQlFilterValue" or similar. Basically, the contains bit is internal to this function.

**Review by aaron-steinfeld — COMMENTED** [review]
overall lgtm

File: projects/observability/src/shared/services/filter-builder/graphql-filter-builder.service.ts (hunk 1 above)
Comment (Christian862) [reply projects/observability/src/shared/services/filter-builder/graphql-filter-builder.service.ts]: good point, fixed

**Review by aaron-steinfeld — APPROVED** [review]

**Review by arjunlalb — DISMISSED** [review]

File: projects/components/src/filtering/filter/filter-operators.ts [hunk 2]
```diff
@@ -9,6 +9,7 @@ export const enum FilterOperator {
   GreaterThanOrEqualTo = '>=',
   Like = '~',
   In = 'IN',
+  Contains = 'CONTAINS',
```
Comment (arjunlalb) [inline projects/components/src/filtering/filter/filter-operators.ts:12]: recommendation : would be helpful to add couple lines describing the differences (or use cases) between LIKE and CONTAINS

Comment (Christian862) [reply projects/components/src/filtering/filter/filter-operators.ts:18]: added

File: projects/components/src/filtering/filter/filter-operators.ts [hunk 3]
```diff
@@ -9,6 +9,12 @@ export const enum FilterOperator {
   GreaterThanOrEqualTo = '>=',
   Like = '~',
   In = 'IN',
+  /**
+   * The difference between 'Contains' and 'Like' is that 'Like' will be treated as a regex filter, not escaping
```
Comment (aaron-steinfeld) [inline projects/components/src/filtering/filter/filter-operators.ts:13]: nit: I would get list into impl details here. CONTAINS is a filter if the field contains the provided string LIKE is a regex match filter.

**Review by aaron-steinfeld — APPROVED** [review]

File: projects/components/src/filtering/filter/filter-operators.ts (hunk 3 above)
Comment (aaron-steinfeld) [reply projects/components/src/filtering/filter/filter-operators.ts:13]: Please follow up on this if you agree

**Pull request merged**